### PR TITLE
BAU: Update PR template to reference acceptance tests and documentation updates

### PR DESCRIPTION
**Pull request by gtvj**
## What

Builds upon recent work to improve our PR template by introducing two additional checklist items:

- the acceptance tests have been updated
- associated documentation (such as the README) has been updated

This change was discussed in the Authentication Developer Meeting on 21 March 2024, where we agreed this would be a good thing to do.

## How to review

1. Review the content and approach

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -37,6 +37,26 @@ You can find example PRs from this repository that include screenshots through t
 
 - [ ] Performance Analysis have been informed of the change
 
+## Acceptance tests have been updated
+
+- [ ] Any necessary changes to the [acceptance tests](https://github.com/govuk-one-login/authentication-acceptance-tests) have been made
```
Comment (BeckaL) [inline .github/PULL_REQUEST_TEMPLATE.md:42]: Are there any general kinds of changes that tend to break acceptance tests the most? I'm thinking like titles of pages, or changes to flow - maybe some other big stuff? Can we chuck them in?

Comment (gtvj) [reply .github/PULL_REQUEST_TEMPLATE.md:42]: Good point - I'll check with the Test Engineers and update this to include them 👍🏽

Comment (gtvj) [reply .github/PULL_REQUEST_TEMPLATE.md]: They've suggested changes to journeys. I had a look at the tests and thought back to things I've encountered so have also included changes to page titles and the text of interactive elements.

**Review by BeckaL — APPROVED** [review]
Thanks for doing this!

**Pull request merged**